packages/core/src/bundle.test.ts

```diff
@@ -415,6 +416,66 @@ describe('Bundle tests', () => {
     });
   });
 
+  describe('reorderBundle', () => {
+    test('Preserves entries without fullUrl and appends them at the end', () => {
+      const inputBundle: Bundle = {
+        resourceType: 'Bundle',
+        type: 'transaction',
+        entry: [
+          // Entry with fullUrl that references Patient (should be reordered)
+          createResourceWithReference('DiagnosticReport', 'urn:uuid:3d8b6e96-6de4-48c1-b7ff-e2c26c924620', {
+            subject: { reference: 'urn:uuid:70653c8f-95e1-4b4e-84e8-8d64c15e4a13' },
+          }),
+          // Entry with fullUrl (should come first after reordering)
+          createResourceWithReference('Patient', 'urn:uuid:70653c8f-95e1-4b4e-84e8-8d64c15e4a13'),
+          // Entry without fullUrl (PATCH operation - should be preserved and appended at end)
+          {
```

Comment:
for a more rigorous test, add the PATCH earlier in the original bundle, and confirm they _still_ end up at the end